docs/csharp/programming-guide/xmldoc/inheritdoc.md

```diff
@@ -0,0 +1,32 @@
+---
+title: "<inheritdoc> - C# Programming Guide"
+ms.date: 01/21/2020
+f1_keywords: 
+  - "inheritdoc"
+  - "<inheritdoc>"
+helpviewer_keywords: 
+  - "<inheritdoc> C# XML tag"
+  - "inheritdoc C# XML tag"
+ms.assetid: 46d329b1-5b84-4537-9e17-73ca97313e4e
+---
+# \<inheritdoc> (C# Programming Guide)
+## Syntax  
+  
+```xml  
+<inheritdoc/> 
+```  
+  
+## InheritDoc  
+Inherit XML comments from base classes, interfaces, and similar methods. This eliminates unwanted copying and pasting of duplicate XML comments and automatically keeps XML comments sychronized. 
+  
+## Remarks  
+Add your XML comments in base classes and let InheritDoc copy the comments to implementing classes.
+
+Add your XML comments in interfaces and let InheritDoc copy the comments to implementing classes.
+
+Add your XML comments to your synchronous methods and let InheritDoc copy the comments to your asynchronous versions of the same methods.  
```

Comment:
As a reader, I'd ask what's the link between the synchronous method and its asynchronous version ?

Is that inferred by the method names ?
So, if I have a two methods called "Connect" and "ConnectAsync", the async one will take the comments from the synchronous one ? Or there is a special way to mark the "Connect" as the synchronous version of "ConnectAsync" ?
What if the methods are "MyConnect" and "ConnectAsync"? How would InheritDoc know that "ConnectAsync" is the async version of "MyConnect" ?

I think the documentation should somehow clarify those.

Reply:
I think the intent here is for this scenario.  Consider this interface:

```csharp
public interface ISomeTaskThings
{
    Task<int> CountAsync();
}
```

The `inheritDoc` attribute would apply on this implementation:

```csharp
public class TaskThingsImpl : ISomeTaskThings
{
    public async Task<int> CountAsync()
    {
        // ...
        return result;
    }
}
```

The addition of the `async` modifier is allowed.

Reply:
More infor on inheritdoc is found here: https://www.inheritdoc.io/

Reply:
@BillWagner I am not sure how to include an example or where the current examples live. I can provide a screenshot but couldnt figure out how to create a snippet.

Reply:
@BillWagner to answer your questions below: 

- Deep Hierarchies: 
The <inheritdoc/> tag works for derived types but not deep hierarchies. There is another syntax for InheritDoc which I wasnt sure where to put <inheritdoc cref ="B"/>. This will work for deep hierarchies. 

![image](https://user-images.githubusercontent.com/46729679/72916773-77a34980-3cf7-11ea-963e-c603434b53a2.png)

- Method overrides and Virtual Methods: 
In general quick Info will now display XML comments for method overrides and virtual methods without InheritDoc so that doc should be updated but not sure exactly where to update that. 

![image](https://user-images.githubusercontent.com/46729679/72916912-bb964e80-3cf7-11ea-8de7-792dc510faf6.png)

![image](https://user-images.githubusercontent.com/46729679/72916957-ce108800-3cf7-11ea-8feb-2a2c3da97cef.png)

- Explicit Interface Implementations:
This will work with the <inheritdoc cref=""/> syntax. 
![image](https://user-images.githubusercontent.com/46729679/72917786-55aac680-3cf9-11ea-96a0-68e6cc0f8abc.png)

- Close generic types:
Not sure about the close generic types. 

- The async example you provided below only works if you have the <inheritdoc cref=""/> tag:

![image](https://user-images.githubusercontent.com/46729679/72917392-9ce48780-3cf8-11ea-8f83-369830b24127.png)